repos/system_upgrade/el7toel8/actors/satelliteupgradefacts/actor.py

```diff
@@ -0,0 +1,133 @@
+from leapp.models import InstalledRPM, Module, RepositoriesSetupTasks, RpmTransactionTasks
+from leapp.tags import FactsPhaseTag, IPUWorkflowTag
+
+
+class SatelliteUpgradeFacts(Actor):
+    """
+    No documentation has been provided for the satellite_upgrade_facts actor.
+    """
+
+    name = 'satellite_upgrade_facts'
+    consumes = (InstalledRPM, )
+    produces = (RepositoriesSetupTasks, RpmTransactionTasks, reporting.Report)
+    tags = (IPUWorkflowTag, FactsPhaseTag)
+
+    def process(self):
+        has_foreman = has_package(InstalledRPM, 'foreman') or has_package(InstalledRPM, 'foreman-proxy')
+        if not has_foreman:
+            return
+
+        local_postgresql = has_package(InstalledRPM, 'rh-postgresql12-postgresql-server')
+        postgresql_contrib = has_package(InstalledRPM, 'rh-postgresql12-postgresql-contrib')
+        postgresql_evr = has_package(InstalledRPM, 'rh-postgresql12-postgresql-evr')
+        # remove packages we don't need anymore
+        self.produce(RpmTransactionTasks(
+            to_remove=['tfm-runtime', 'tfm-pulpcore-runtime', 'rh-redis5-runtime', 'rh-ruby27-runtime', 'rh-python38-runtime'],
+            # you'll need to define the repos in leapp_upgrade_repositories.repo
+            # until this is on the CDN
+            to_install=['rubygem-foreman_maintain', 'foreman-installer-katello'],
+            modules_to_enable=[Module(name='ruby', stream='2.7'), Module(name='python38', stream='3.8')]
+            )
+        )
+
+        if has_package(InstalledRPM, 'katello'):
+            self.produce(RpmTransactionTasks(
+                modules_to_enable=[Module(name='pki-core', stream='10.6'), Module(name='pki-deps', stream='10.6')],
+                to_install=['katello', 'katello-selinux']
+                )
+            )
+
+        if has_package(InstalledRPM, 'rh-redis5-redis'):
+            self.produce(RpmTransactionTasks(
+                modules_to_enable=[Module(name='redis', stream='5')],
+                to_install=['redis']
+                )
+            )
+
+        additional_rpms = []
+        for rpm_pkgs in self.consume(InstalledRPM):
+            for pkg in rpm_pkgs.items:
+                if pkg.name.startswith('tfm-rubygem-hammer') or pkg.name.startswith('tfm-rubygem-foreman') or pkg.name.startswith('tfm-rubygem-katello') or pkg.name.startswith('tfm-rubygem-smart_proxy'):
+                    additional_rpms.append(pkg.name.replace('tfm-rubygem-', 'rubygem-'))
+                elif pkg.name.startswith('tfm-pulpcore-python3-pulp'):
+                    additional_rpms.append(pkg.name.replace('tfm-pulpcore-python3-', 'python38-'))
+                elif pkg.name.startswith('foreman-installer') or pkg.name.startswith('satellite-installer'):
+                    additional_rpms.append(pkg.name)
+        self.produce(RpmTransactionTasks(to_install=additional_rpms))
+
+        if local_postgresql:
+            scl_psql_path = '/var/opt/rh/rh-postgresql12/lib/pgsql/data/'
```

Comment:
There's a `POSTGRESQL_SCL_DATA_PATH` constant in the other actor. Could this be unified?

Reply:
Technically you could create a shared library to share code between 2 actors (like `repos/system_upgrade/el7toel8/libraries/satelliteupgradeutils.py` and then import it as `import leapp.libraries.common.satelliteupgradeutils as lib`). But for one single constant that is used for report generation only this may be an overkill, so up to you.

Anyway, could you please make this string a constant 
`POSTGRESQL_SCL_DATA_PATH = '/var/opt/rh/rh-postgresql12/lib/pgsql/data/'` as well?